src/main/java/org/vaadin/application/model/Invoice.java

```diff
@@ -0,0 +1,161 @@
+package org.vaadin.application.model;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.DecimalMin;
+
+import java.time.LocalDate;
+import java.util.UUID;
+
+/**
+ * Entity representing an invoice.
+ * An invoice has a recipient name, an amount payable, and additional details like issue date, etc.
+ */
+@Entity
+@Table(name = "invoice")
+public class Invoice {
+
+    /**
+     * The unique identifier for the invoice.
+     */
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private UUID id;
+
+    /**
+     * The unique number of the invoice.
+     * Cannot be null.
+     */
+    @NotNull
+    private String invoiceNumber;
+
+    /**
+     * The name of the recipient for the invoice.
+     * Cannot be null.
+     */
+    @NotNull
+    private String recipientName;
+
+    /**
+     * The amount payable for the invoice.
+     * Must be a positive value greater than 0.0.
+     */
+    @NotNull
+    @DecimalMin(value = "0.0", inclusive = false)
+    private double amount;
+
+    /**
+     * The issue date of the invoice.
+     * Cannot be null.
+     */
+    @NotNull
+    private LocalDate issueDate;
```

Comment:
I noticed other models such as `Expense` and `Income` use the `Date` class to represent the date instead of `LocalDate`. This could be changed so it is consistent across the models however it is not a critical issue so its entirely up to you :)

Reply:
Thank you for pointing this out! Field types have been updated to use the Date class.